elementary-planner: hardcoded themes & vala 0.56 fixes

bobby285271 opened the pull request:
###### Description of changes

- Fixes the wrong hardcoded GTK theme name
- Tries to address #163623
- Fixes build with vala 0.56

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

bobby285271 committed:
elementary-planner: fix build with vala 0.56

bobby285271 commented:
@ofborg eval

jtojnar commented on pkgs/applications/office/elementary-planner/default.nix:
```diff
@@ -72,6 +83,8 @@ stdenv.mkDerivation rec {
     gappsWrapperArgs+=(
       # the theme is hardcoded
       --prefix XDG_DATA_DIRS : "${pantheon.elementary-gtk-theme}/share"
+      # the icon is hardcoded
```
```suggestion
      # the icon theme is hardcoded
```

bobby285271 committed:
elementary-planner: fix wrong hardcoded GTK theme name

The theme name is updated in elementary OS 6.

bobby285271 committed:
elementary-planner: pick up hardcoded icon theme

With https://github.com/NixOS/nixpkgs/pull/163623, we need to handle
the hardcoded icon theme ourselves.

Co-authored-by: Jan Tojnar <jtojnar@gmail.com>

bobby285271 commented:
Result of `nixpkgs-review pr 164555` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>elementary-planner</li>
  </ul>
</details>
